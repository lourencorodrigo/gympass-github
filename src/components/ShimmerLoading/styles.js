import styled from "styled-components";

import { em } from "../../styles/tools";

export const Wrapper = styled.div`
  width: ${({ width }) => em(width)};
  height: ${({ height }) => em(height)};
  background: #252b38;
  border-radius: ${em(8)};
`;
