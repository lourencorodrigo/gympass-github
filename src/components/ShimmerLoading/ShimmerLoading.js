import React from "react";
import PropTypes from "prop-types";

import { Wrapper } from "./styles";

const ShimmerLoading = ({ width = "250", height = "16" }) => (
  <Wrapper width={width} height={height} />
);

ShimmerLoading.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number
};

export default ShimmerLoading;
