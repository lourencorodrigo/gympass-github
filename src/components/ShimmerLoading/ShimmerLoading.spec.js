import React from "react";
import ReactDOM from "react-dom";
import ReactTestRenderer from "react-test-renderer";

import ShimmerLoading from "./";

describe("ShimmerLoading component", () => {
  const element = <ShimmerLoading />;

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(element, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("should compare the component with a snapshot", () => {
    const component = ReactTestRenderer.create(element);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
