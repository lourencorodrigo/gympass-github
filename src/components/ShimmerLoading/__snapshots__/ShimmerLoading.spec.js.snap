// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShimmerLoading component should compare the component with a snapshot 1`] = `
<div
  className="sc-bdVaJa dpTItc"
  height="16"
  width="250"
/>
`;
